core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -487,6 +487,14 @@ private boolean expandSelectItem(final SqlNode selectItem, SqlSelect select,
       // calls.
       selectScope = getSelectScope(select);
       expanded = expandSelectExpr(selectItem, scope, select, expansions);
+
+      // Non-strict GROUP BY: wrap non-aggregated, non-grouped columns in ANY_VALUE()
+      if (isAggregate(select)
+          && config.conformance().isNonStrictGroupBy()
+          && isNonAggregatedNonGroupedColumn(expanded, select)) {
```

Comment:
I wonder how this interacts with other config options such as isGroupByAlias or isSelectAlias.
But perhaps this is fine, since this check is after expandSelectExpr.